### PR TITLE
Misc changes from CR with Tim/Abhipsa/Nick/Nate

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -164,11 +164,11 @@ dotnet_naming_rule.interface_should_be_begins_with_i.style = begins_with_i
 
 dotnet_naming_rule.types_should_be_pascal_case.severity = suggestion
 dotnet_naming_rule.types_should_be_pascal_case.symbols = types
-dotnet_naming_rule.types_should_be_pascal_case.style = pascal_case
+dotnet_naming_rule.types_should_be_pascal_case.style = non_private_static_field_style
```
Comment (drwill-ms) [inline .editorconfig:167]: Opening the editorconfig made a bunch of changes, and we changed one thing for protected internal fields to format the way we expect.

If you see any IDE recommendations for style that don't match what we expect, let me know.

File: e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs [hunk 2]
```diff
@@ -24,26 +24,22 @@ public class PurgeMesageQueueE2ETests : E2EMsTestBase
...
         {
             using TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
-            var deviceId = testDevice.Device.Id;
+            string expectedDeviceId = testDevice.Device.Id;
             using var sc = new IotHubServiceClient(TestConfiguration.IotHub.ConnectionString);
-            PurgeMessageQueueResult result = await sc.Messages.PurgeMessageQueueAsync(deviceId, CancellationToken.None).ConfigureAwait(false); // making sure the queue is empty
+            PurgeMessageQueueResult result = await sc.Messages.PurgeMessageQueueAsync(expectedDeviceId, CancellationToken.None).ConfigureAwait(false); // making sure the queue is empty
 
             Message testMessage = ComposeD2CTestMessage();
-            var expectedResult = new PurgeMessageQueueResult
```
Comment (drwill-ms) [inline e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs:32]: Unnecessary.

File: iothub/device/src/IotHubBaseClient.cs [hunk 3]
```diff
@@ -36,12 +35,7 @@ public abstract class IotHubBaseClient : IDisposable
         private object _twinPatchCallbackContext;
         private Func<TwinCollection, object, Task> _desiredPropertyUpdateCallback;
 
-        // Diagnostic information
-
-        // Count of messages sent by the device/ module. This is used for sending diagnostic information.
-        private int _currentMessageCount;
-
-        private int _diagnosticSamplingPercentage;
+        private protected readonly IotHubClientOptions _clientOptions;
```
Comment (drwill-ms) [inline iothub/device/src/IotHubBaseClient.cs:38]: Doesn't need to be a property.

File: iothub/device/src/IotHubBaseClient.cs [hunk 4]
```diff
@@ -36,12 +35,7 @@ public abstract class IotHubBaseClient : IDisposable
         private object _twinPatchCallbackContext;
         private Func<TwinCollection, object, Task> _desiredPropertyUpdateCallback;
 
-        // Diagnostic information
-
-        // Count of messages sent by the device/ module. This is used for sending diagnostic information.
-        private int _currentMessageCount;
```
Comment (drwill-ms) [inline iothub/device/src/IotHubBaseClient.cs:42]: Removed along with some other diagnostics code that is really old and we don't think is important (doesn't exist in Java client, etc.).

File: iothub/device/src/IotHubClientOptions.cs [hunk 5]
```diff
@@ -66,27 +66,28 @@ public IotHubClientOptions(IotHubClientTransportSettings transportSettings)
         /// <summary>
         /// The DTDL model Id associated with the device or module client instance.
         /// </summary>
+        /// <remarks>
         /// This feature is currently supported only over MQTT and AMQP transports.
-        /// <remarks></remarks>
```
Comment (drwill-ms) [inline iothub/device/src/IotHubClientOptions.cs:70]: Oops

File: iothub/device/src/IotHubClientOptions.cs [hunk 6]
```diff
@@ -66,27 +66,28 @@ public IotHubClientOptions(IotHubClientTransportSettings transportSettings)
...
+        /// The default behavior is that <see cref="Message.MessageId"/> is set only by the user.
         /// </remarks>
         public SdkAssignsMessageId SdkAssignsMessageId { get; set; } = SdkAssignsMessageId.Never;
 
         /// <summary>
-        /// Stores custom product information that will be appended to the user agent string that is sent to IoT hub.
+        /// Specifies additional information that will be appended to the user-agent string that is sent to IoT hub.
         /// </summary>
-        public string CustomProductInfo
+        public string AdditionalUserAgentInfo
```
Comment (drwill-ms) [inline iothub/device/src/IotHubClientOptions.cs:85]: Rename to make more sense to users.

File: iothub/device/src/IotHubDeviceClient.cs [hunk 7]
```diff
@@ -183,7 +183,7 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
-        internal override void AddToPipelineContext()
+        private protected override void AddToPipelineContext()
```
Comment (drwill-ms) [inline iothub/device/src/IotHubDeviceClient.cs:186]: private protected is another special combo, like protected internal. It means only to types in this library that inherit from me.

File: iothub/service/src/Jobs/Models/ScheduledJob.cs [hunk 8]
```diff
@@ -12,12 +12,22 @@ namespace Microsoft.Azure.Devices
     /// </summary>
     public class ScheduledJob : IotHubScheduledJobResponse
     {
+        /// <summary>
+        /// This constructor is for deserialization and unit test mocking purposes.
```
Comment (drwill-ms) [inline iothub/service/src/Jobs/Models/ScheduledJob.cs:16]: heads up @patilsnr and @ngastelum-ms

File: iothub/device/src/IotHubBaseClient.cs [hunk 9]
```diff
@@ -483,6 +450,8 @@ internal void OnDesiredStatePatchReceived(TwinCollection patch)
             _ = _desiredPropertyUpdateCallback.Invoke(patch, _twinPatchCallbackContext);
         }
 
+        private protected abstract void AddToPipelineContext();
```
Comment (abhipsaMisra) [inline iothub/device/src/IotHubBaseClient.cs:453]: +1

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/device/src/IotHubBaseClient.cs (hunk 9 above)
Comment (timtay-microsoft) [reply iothub/device/src/IotHubBaseClient.cs:453]: This method will disappear shortly when we consolidate the setMessageReceiveHandler code, FYI

**Review by andyk-ms — APPROVED** [review]

**Pull request merged**